linmod/eval.py

```diff
@@ -1,124 +1,210 @@
+import numpy as np
 import polars as pl
 
 from linmod.utils import pl_list_cycle, pl_norm
 
 
-def _merge_samples_and_data(samples, data):
-    r"""
-    Join the forecast samples and raw data dataframes, assuming they have the
-    standard format.
-
-    Also compute the true proportions from the raw data.
-    """
-    result = (
-        data.with_columns(
-            phi=(
-                pl.col("count") / pl.sum("count").over("fd_offset", "division")
-            ),
-        )
-        .drop("count")
-        .join(
-            samples,
-            on=("fd_offset", "division", "lineage"),
-            how="left",
-            suffix="_sampled",
+def multinomial_count_sampler(N, phi, rng):
+    return rng.multinomial(N, phi)
+
+
+class ProportionsEvaluator:
+    def __init__(self, samples: pl.LazyFrame, data: pl.LazyFrame):
+        """
+        `samples` should have the standard model output format.
```

Comment:
Do you have any validation methods, so that you can check up front that these df's have the schema you expect?

might not be necessary, but it's a few lines of code that can save a lot of headache later

Reply:
Yeah, I think as this codebase grows, I'm seeing the merits of something like the `MultinomialModelOutput` you wrote in #67.

Does this belong in this PR or a new one?

Reply:
New PR!

Reply:
Will be done in new PR